documentation/src/test/java/org/hibernate/search/documentation/search/predicate/FieldReferenceIT.java

```diff
@@ -0,0 +1,589 @@
+
+			public e1_() {
+				a = ValueFieldReference1.of( "e1.a", ContainingA__.class, String.class, String.class, String.class );
+				b = ValueFieldReference1.of( "e1.b", ContainingA__.class, String.class, String.class, String.class );
+			}
+
+			@Override
+			public String absolutePath() {
+				return "e1";
+			}
+
+			@Override
+			public Class<ContainingA__> scopeRootType() {
+				return ContainingA__.class;
+			}
+
+			@Override
+			public ValueFieldReference1<ContainingA__, String, String, String> a() {
+				return a;
+			}
+		}
+
+		public static class e2_
+				implements ObjectFieldReference<ContainingA__>,
+				ContainingA_e1_e2_Intersection<ContainingA__>,
+				ContainingA_e1_e2_e3_Intersection<ContainingA__> {
+
+			public final ValueFieldReference1<ContainingA__, String, String, String> a;
+			public final ValueFieldReference1<ContainingA__, String, String, String> b;
+
+			public e2_() {
+				a = ValueFieldReference1.of( "e2.a", ContainingA__.class, String.class, String.class, String.class );
+				b = ValueFieldReference1.of( "e2.b", ContainingA__.class, String.class, String.class, String.class );
+			}
+
+			@Override
+			public String absolutePath() {
+				return "e2";
+			}
+
+			@Override
+			public Class<ContainingA__> scopeRootType() {
+				return ContainingA__.class;
+			}
+
+			@Override
+			public ValueFieldReference1<ContainingA__, String, String, String> a() {
+				return a;
+			}
+		}
+
+		public static class e3_
+				implements ObjectFieldReference<ContainingA__>,
+				ContainingA_e1_e2_e3_Intersection<ContainingA__> {
+
+			public final ValueFieldReference2<ContainingA__, String, String> a;
+
+			public e3_() {
+				a = ValueFieldReference2.of( "e3.a", ContainingA__.class, String.class, String.class );
+			}
+
+			@Override
+			public String absolutePath() {
+				return "e3";
+			}
+
+			@Override
+			public Class<ContainingA__> scopeRootType() {
+				return ContainingA__.class;
+			}
+
+			@Override
+			public ValueFieldReference2<ContainingA__, String, String> a() {
+				return a;
+			}
+		}
+	}
+
+	public interface ContainingA_e1_e2_e3_Intersection<SR> {
+
+		TraitsIntersection<SR, String> a();
+
+		interface TraitsIntersection<SR, T>
+				extends MatchPredicateFieldReference<SR, T>,
+				ExistsPredicateFieldReference<SR>,
+				QueryStringPredicateFieldReference<SR, T> {
+
+			@Override
+			default ValueConvert valueConvert() {
+				return ValueConvert.YES;
+			}
+		}
+	}
+
+	// the same field with the same traits:
+	public interface ContainingA_e1_e2_Intersection<SR> {
+
+		TraitsIntersection<SR, String, String> a();
+
+		interface TraitsIntersection<SR, T, P>
+				extends FieldProjectionFieldReference<SR, P>,
+				MatchPredicateFieldReference<SR, T>,
+				ExistsPredicateFieldReference<SR>,
+				QueryStringPredicateFieldReference<SR, T> {
+
+			@Override
+			default ValueConvert valueConvert() {
+				return ValueConvert.YES;
+			}
+		}
+	}
```

Comment:
Are you sure there's a need to have special handling for the `TraitsInteresection` interface?

I mean, we already need to generate interfaces/classes for combination of traits, so we already assume something will be generating a `FieldProjectionAndMatchPredicateAndExistsPredicateAndQueryStringPredicateFieldReference`; can't we use that here, too?

Unrelated note: naming these "trait combination" interfaces will be a challenge in itself.

Reply:
> Are you sure there's a need to have special handling for the TraitsInteresection interface?

thinking more about it, yeah we can probably reuse the interfaces... I was thinking about a scenario:

```java
@KeywordField(projectable = Projectable.YES, sortable = Sortable.NO)
private String title;

@KeywordField(projectable = Projectable.NO, sortable = Sortable.YES)
private String title;
```
where the intersection would be just predicates and we wouldn't have it since it is not needed for an actual field, but we could just keep creating these combinations and add them to a lookup for reuse, so +1 😃 

> Unrelated note: naming these "trait combination" interfaces will be a challenge in itself.

yeah... At some point I was thinking to have a sorted list of trait interfaces and then "map" them to some numbered keys, e.g. T1, T2... Tn, and the name of a combination will be based on these keys, e.g. `T1T5T24FieldReferece`